packages/zoe/src/zoeService/startInstance.js

```diff
@@ -43,13 +50,55 @@ export const makeStartInstance = (
     seatHandleToZoeSeatAdmin,
   );
 
+  const getFreshZcfBundleCap = async () => {
+    const settledBundleCap = await getZcfBundleCapP();
+    settledBundleCap !== undefined || Fail`the ZCF bundle cap was broken`;
+    return settledBundleCap;
+  };
+
   const InstanceAdminStateShape = harden({
     instanceStorage: M.remotable('ZoeInstanceStorageManager'),
     instanceAdmin: M.remotable('InstanceAdmin'),
-    seatHandleToSeatAdmin: M.remotable(),
+    seatHandleToSeatAdmin: M.remotable(), // seatHandleToSeatAdmin, but putting that string here is backwards-incompatible
     adminNode: M.remotable('adminNode'),
   });
 
+  /** @type {{ onFulfilled: (completion: any, instanceAdmin: InstanceAdmin) => void, onRejected: (reason: any, instanceAdmin: InstanceAdmin, adminNode: any) => void }} */
+  const watcher = prepareExo(
+    zoeBaggage,
+    'InstanceCompletionWatcher',
+    M.interface('InstanceCompletionWatcher', {
+      onFulfilled: M.call(
+        M.any(),
+        InstanceAdminShape,
+        M.remotable('adminNode'),
+      ).returns(),
+      onRejected: M.call(
+        M.any(),
+        InstanceAdminShape,
+        M.remotable('adminNode'),
+      ).returns(),
+    }),
+    {
+      onFulfilled: (completion, instanceAdmin) =>
+        instanceAdmin.exitAllSeats(completion),
+      onRejected: (/** @type {Error} */ reason, instanceAdmin, adminNode) => {
+        if (isUpgradeDisconnection(reason)) {
+          console.log(`resetting promise watcher after upgrade`, reason);
+          // eslint-disable-next-line no-use-before-define
+          watchForAdminNodeDone(adminNode, instanceAdmin);
```

Comment:
I'd prefer `watchForAdminNodeDone` to be part of the watcher for less circularity, but this does work.

Reply:
It's a separate function because it's called from other places outside the watcher.

Reply:
Out of scope of for this PR